AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/ViewNotifierHelper.java

```diff
@@ -49,11 +58,25 @@ public JVar replacePreviousNotifier(JBlock block) {
 		AbstractJClass notifierClass = holder.getEnvironment().getJClass(OnViewChangedNotifier.class);
 		if (notifier == null) {
 			notifier = holder.getGeneratedClass().field(PRIVATE | FINAL, notifierClass, "onViewChangedNotifier" + generationSuffix(), _new(notifierClass));
-			holder.getGeneratedClass()._implements(HasViews.class);
+			implementHasViewsInHolder();
 		}
 		return block.decl(notifierClass, "previousNotifier", notifierClass.staticInvoke("replaceNotifier").arg(notifier));
 	}
 
+	private void implementHasViewsInHolder() {
+		holder.getGeneratedClass()._implements(HasViews.class);
+		JCodeModel codeModel = environment.getCodeModel();
+
+		JDirectClass genericType = codeModel.directClass("T");
+		JMethod findViewById = holder.getGeneratedClass().method(PUBLIC, genericType, "internalFindViewById");
+		findViewById.generify("T", environment.getClasses().VIEW);
+		findViewById.annotate(Override.class);
+
+		JVar idParam = findViewById.param(codeModel.INT, "id");
+		IJExpression findViewByIdExpression = holder.getFindViewByIdExpression(idParam);
+		findViewById.body()._return(JExpr.cast(genericType, findViewByIdExpression));
```

Comment:
Well, it would be nice to gain something from this new signature, if we doing so much work because of it. 😆 Since you already have the check code, i guess you can add it now.